weave/tests/test_index.py

```diff
@@ -1262,6 +1314,59 @@ def test_index_abc_get_children_complex_fail(test_pantry):
         ind.get_children(parent_path)
 
 
+def test_index_abc_get_children_min_gen_level(test_pantry):
+    """Test IndexABC get_children min_gen_level removes uuids where the
+    gen_level is less than the min_gen_level.
+    """
+
+    # Unpack the test_pantry into two variables for the pantry and index.
+    test_pantry, ind = test_pantry
+
+    # Setup random strucutre of parents and children
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3000")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3003")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_2")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3333")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_3")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3303")
+
+    tmp_dir = test_pantry.set_up_basket("grandparent_2")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="2000", parent_ids=["3000", "3003", "3333"]
+    )
+
+    tmp_dir = test_pantry.set_up_basket("grandparent_2_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="2002")
+
+    tmp_dir = test_pantry.set_up_basket("parent_1")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="1000", parent_ids=["2000", "2002", "3303"]
+    )
+
+    tmp_dir = test_pantry.set_up_basket("parent_1_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="1001")
+
+    tmp_dir = test_pantry.set_up_basket("child_0")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="0000", parent_ids=["1001", "1000"]
+    )
+
+    ind.generate_index()
+
+    children_df = ind.get_children("3000", min_gen_level=-2)
+    # Create a dataframe with min_gen_level filtered. This should be exactly
+    # equal to the children_df if min_gen_level works correctly.
+    min_gen_level_children_df = children_df[
+                                children_df["generation_level"] >= -2]
+
+    assert children_df.equals(min_gen_level_children_df)
```

Comment:
I think this test could miss a lot of cases. We should discuss it.

Reply:
What are you thinking that it could miss? I can't think of anything.

Reply:
I think the assumption is you're creating the min_gen_level_children_df from the result of get_children. Think of the case where get_children doesn't return anything, or any subset of the correct dataframe. This assert will always return true even if it is missing correct answers.

I think your previous test was more comprehensive. You could just convert the child uuids to a list, then check make sure that list has all the uuids you want, and only those.

Reply:
I updated those two tests if you wanted to take a look at them.